project/cmake/scripts/common/macros.cmake

```diff
@@ -62,20 +62,20 @@ endfunction()
 #   (if NO_INSTALL is not given).
 function(copy_file_to_buildtree file relative)
   cmake_parse_arguments(arg "NO_INSTALL" "" "" ${ARGN})
-  if(NOT WIN32)
-    string(REPLACE "\(" "\\(" file ${file})
-    string(REPLACE "\)" "\\)" file ${file})
-  endif()
   string(REPLACE "${relative}/" "" outfile ${file})
+  get_filename_component(outdir ${outfile} DIRECTORY)
 
   if(NOT TARGET export-files)
-    add_custom_target(export-files ALL COMMENT "Copying files into build tree")
+    file(REMOVE ${CMAKE_BINARY_DIR}/${CORE_BUILD_DIR}/ExportFiles.cmake)
+    add_custom_target(export-files ALL COMMENT "Copying files into build tree"
+                      COMMAND ${CMAKE_COMMAND} -P ${CMAKE_BINARY_DIR}/${CORE_BUILD_DIR}/ExportFiles.cmake)
   endif()
   if(NOT ${CORE_SOURCE_DIR} MATCHES ${CMAKE_BINARY_DIR})
     if(VERBOSE)
       message(STATUS "copy_file_to_buildtree - copying file: ${file} -> ${CMAKE_CURRENT_BINARY_DIR}/${outfile}")
     endif()
-    add_custom_command(TARGET export-files COMMAND ${CMAKE_COMMAND} -E copy_if_different "${file}" "${CMAKE_CURRENT_BINARY_DIR}/${outfile}")
+    file(APPEND ${CMAKE_BINARY_DIR}/${CORE_BUILD_DIR}/ExportFiles.cmake
+         "file(COPY \"${file}\" DESTINATION \"${outdir}\")\n")
```

Comment:
I wanted to ping you after I figured out what jenkins has to complain about on windows, but nice that you watch these things!

It was a good chance to play with https://github.com/volo-zyko/cmake-profile-stats for profiling and that works well (after patching cmake). I've tried a few different things that are probably worth documenting:
- `file(COPY)` directly at configure time: cmake is not always rerun and it won't work if you just touch a file and nothing else
- `configure_file` directly at configure time: works and is slightly faster than `copy_if_different`
- add a template ExportFile.cmake and have one custom command per file calling it: similar performance to `copy_if_different`

It still seems that `file(APPEND` is not optimal on Windows as it causes the cmake run to be a bit slower. But I don't have a better idea atm, and it already significantly works faster.

(new windows jenkins build: http://jenkins.kodi.tv/job/WIN-32/8517/)